src/exanb/grid_cell_particles/include/exanb/grid_cell_particles/lattice_generator.h

```diff
@@ -94,6 +97,96 @@ namespace exanb
   public:
     inline void execute () override final
     {
+      // Checking the required lattice :
+      //   if 'CUSTOM' -> check required fields then create LatticeCollection
+      //   if one of predefined lattices 'SC' 'BCC' 'FCC' 'HCP' etc... -> check required fields then create Lattice Collection
+
+      LatticeCollection lattice;
+      lattice.m_structure = *structure;
+      lattice.m_size = *size;
+      if (*structure == "CUSTOM") {
+        assert( (*np).has_value() );
+        assert( (*positions).has_value() );
+        lattice.m_np = *np;
+        lattice.m_types = *types;
+        // Checking that positions are contained between 0 and 1 -> fractional coordinates
+        for (int i=0;i<(*positions).size();i++) {
+          double px = (*positions)[i].x;
```

Comment:
Add `[[maybe_unused]]` here for compiler or define this section `#ifndef NDEGUB` (avoid warning)

px, py, pz